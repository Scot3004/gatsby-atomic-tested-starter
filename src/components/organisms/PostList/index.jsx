import React from 'react'
import PropTypes from 'prop-types'
import PostPreview from '../../molecules/PostPreview'

export const PostList = ({ data }) => (
  <>
    {data.map((post, i) => (
      <PostPreview
        key={i}
        title={post.node.frontmatter.title}
        image={post.node.frontmatter.image}
        excerpt={post.node.excerpt}
        slug={post.node.fields.slug}
      />
    ))}
  </>
)

PostList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    node: PropTypes.shape({
      excerpt: PropTypes.string,
      frontmatter: PropTypes.shape({
        image: PropTypes.string,
        title: PropTypes.string
      }),
      fields: PropTypes.shape({
        slug: PropTypes.string
      })
    })
  }))
}

export default PostList
