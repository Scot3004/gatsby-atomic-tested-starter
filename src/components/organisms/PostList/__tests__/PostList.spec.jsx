import React from 'react';
import 'jest-styled-components';
import ShallowRenderer from 'react-test-renderer/shallow';
import PostList from '..';

const stub = [
  {
    node: {
      id: '39cb6d6b-1bf3-5aa1-a310-b882314db77b',
      excerpt:
        '"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi utâ€¦',
      frontmatter: {
        date: 'November 07, 2017',
        title: 'My first blog post',
        image: 'https://www.secorto.com/img/logo.png'
      },
      fields: {
        slug: '/blog/my-first-post'
      }
    }
  }
];

describe('When I want to see the list of posts', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<PostList data={stub} />);
  test('Then the list of posts should be displayed', () => {
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
