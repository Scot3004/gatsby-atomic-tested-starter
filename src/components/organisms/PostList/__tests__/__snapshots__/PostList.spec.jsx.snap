// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c1 {
  -webkit-flex: 0 0 64px;
  -ms-flex: 0 0 64px;
  flex: 0 0 64px;
  width: 64px;
  height: 64px;
  padding: 1rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 0 1rem 0;
  padding: .6rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  width: 100%;
}

.c0:hover {
  box-shadow: 0 6px 12px 0 rgba(0,0,0,0.2);
}

.c0:last-child {
  margin: 0;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c4 {
  margin: 0;
}

.c3 {
  margin: 0 0 .6rem 0;
  padding: 0;
  word-wrap: none;
}

<div
  className="c0"
>
  <img
    alt="My first blog post"
    className="c1"
    size="64px"
    src="https://www.secorto.com/img/logo.png"
  />
  <div
    className="c2"
  >
    <a
      href="/blog/my-first-post"
    >
      <h2
        className="c3"
      >
        My first blog post
      </h2>
    </a>
    <p
      className="c4"
    >
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi utâ€¦
    </p>
  </div>
</div>
`;
